cfme/common/vm.py

```diff
@@ -442,6 +442,31 @@ def unset_ownership(self):
         view = self.create_view(navigator.get_class(self, 'Details').VIEW)
         view.flash.assert_success_message(msg)
 
+    def rename_vm(self, new_vm_name, cancel=False, reset=False):
+        """Rename the VM
+
+        Args:
+            new_vm_name: object for renaming vm
+            cancel (bool): Whether to cancel form submission
+            reset (bool): Whether to reset form after filling
+        """
+        view = navigate_to(self, 'Rename')
+        changed = view.vm_name.fill(new_vm_name)
+        if changed:
+            if reset:
+                view.reset_button.click()
+                view.flash.assert_no_error()
+                view.cancel_button.click()
+            else:
+                # save the form
+                view.save_button.click()
+                view.flash.assert_no_error()
+                return self.appliance.collections.infra_vms.instantiate(new_vm_name, self.provider,
```

Comment:
Why instantiate a new VM instead of just setting `self.name = new_vm_name`?

Reply:
Yes, I have updated it with your suggestion. As we are only changing name of VM. So we don't need to reinitialize whole VM object again.
Thanks for this suggestion.